### PR TITLE
feat(a380x/nd): support 640nm range

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

* Refactor shared ND to support specifying allowed range values in props
* Specify A320 ND range values
* Specify A380 ND range values
* Fix up legacy FMGC build typings

## Testing instructions

* Ensure ND EFIS filters work correctly
* Ensure ND range switching works correctly

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by 2hwk** [comment]
Note that lint is failing 
```
/external/fbw-common/src/systems/instruments/src/ND/ND.tsx
  153:56  error  'value' is defined but never used. Allowed unused args must match /^_|^node$|^deltaTime$/u  @typescript-eslint/no-unused-vars
```

File: fbw-a32nx/src/systems/fmgc/src/guidance/GuidanceController.ts [hunk 1]
```diff
@@ -99,11 +95,11 @@ export class GuidanceController {
 
     automaticSequencing: boolean = true;
 
-    leftEfisState: EfisState
+    leftEfisState: EfisState<any>
```
Comment (flogross89) [inline fbw-a32nx/src/systems/fmgc/src/guidance/GuidanceController.ts:98]: why EfisState<any>, and not EfisState<number>? 
Otherwise, LGTM.

Comment (tracernz) [reply fbw-a32nx/src/systems/fmgc/src/guidance/GuidanceController.ts:98]: Or EfisState<unknown>.

Comment (Benjozork) [reply fbw-a32nx/src/systems/fmgc/src/guidance/GuidanceController.ts]: `unknown` isn't accepted by typescript here unfortunately

**Review by flogross89 — APPROVED** [review]
LGTM

**Comment by Popespice** [comment]
Quality Assurance Report

Discord          : Popespice
Object of testing: #8450 
Tier of Testing  : 1
Date             : 08022024

Testing Process: 
1. Log into sim
2. Load into A320NX
3. Power on aircraft as usual
4. Adjust ND range all the way out to max
5. Observe 320 Max range is accurate.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
Working as expected

Media:       
N/A

**Review by tracernz — APPROVED** [review]

**Pull request merged**